docs/components/file-input-field/index.md

```diff
@@ -0,0 +1,106 @@
+# File Input Field
+
+Provides an underlying `<input type="file">` element building on top of the Field component.
+
+## Label
+
+Provide a string to `@label` to render the text into the `<input type="file">` of the Field.
+
+## Hint
+
+Provide a string to `@hint` to render the text into the Hint section of the Field. This is optional.
+
+## Error
+
+Provide a string to `@error` to render the text into the Error section of the Field. This is optional.
+
+## files
+
+Provide a list of [File](https://developer.mozilla.org/en-US/docs/Web/API/File) objects to the File input.
+
+**:lightbulb: Note:** To make things easier, the `@files` argument is an array. This makes it convenient to use existing Array methods like `find` and `filter`.
+
+However, the `files` from the `event.target` in the `@onChange` callback is a FileList.
+
+To convert from a `FileList` to an array of `Files`, use the `...` (spread) operator.
+
+## Value and onChange
+
+To tie into the input event, provide `@onChange`. `@onChange` will take a FileEvent argument, which should contain an `event.target.files` which is a [FileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList).
+
+```hbs
+<Form::FileInputField
+  @label='Label'
+  @files={{this.files}}
+  @onChange={{this.handleChange}}
+/>
+```
+
+```js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class extends Component {
+  // Note that a File object is different from a FileList!
+  @tracked files: File[] = [];
+
+  @action
+  handleChange(e) {
+    console.log({ files: e.target.files });
+    // convert to a File[] by using the spread operator
```

Comment:
Added comment here about using `...` (spread) operator.

Reply:
I do not understand this, `files` is already of type `File[]`, isn't?